### PR TITLE
Migrate dunesql & support new chains for `layerzero`

### PR DESCRIPTION
**Pull request by BennyFeng**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

**Comment by BennyFeng** [comment]
recheck

**Comment by BennyFeng** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by BennyFeng** [comment]
@jeff-dude I tried to migrate to DuneSQL and add support for new chains. I think [the layerzero issue](https://github.com/duneanalytics/spellbook/issues/4102) can be fixed by DuneSQL, although I can't replicate it in DuneSQL(Sparksql option has been hidden).

**Comment by jeff-dude** [comment]
> @jeff-dude I tried to migrate to DuneSQL and add support for new chains. I think [the layerzero issue](https://github.com/duneanalytics/spellbook/issues/4102) can be fixed by DuneSQL, although I can't replicate it in DuneSQL(Sparksql option has been hidden).

amazing -- thanks for the quick response and build here!

i forgot you wouldn't be able to debug on spark, my apologies for that. i wasn't able to recreate on spark anyway. i think a fresh start here is for the best.

i'll get on the review asap

File: models/layerzero/arbitrum/layerzero_arbitrum_send.sql [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {{ config(
+    tags=['dunesql'],
     alias = alias('send'),
     partition_by = ['block_date'],
```
Comment (jeff-dude) [inline models/layerzero/arbitrum/layerzero_arbitrum_send.sql:4]: we've been looking to increase our partition size -- can you introduce a new column called `block_month`, which mimics `block_date` but truncates at the monthly level? then apply column to partition?

this applies to all blockchain version spells here

File: models/layerzero/arbitrum/layerzero_arbitrum_send.sql [hunk 2]
```diff
@@ -24,30 +25,30 @@ WITH send_detail AS (
...
+        AND t.block_time >= date('{{transaction_start_date}}')
         {% endif %}
         {% if is_incremental() %}
-        AND t.block_time >= date_trunc("day", now() - interval '1 week')
+        AND t.block_time >= date_trunc('day', now() - interval '7' day)
         {% endif %}
     WHERE s.call_success
         {% if not is_incremental() %}
-        AND s.call_block_time >= '{{transaction_start_date}}'
+        AND s.call_block_time >= date('{{transaction_start_date}}')
```
Comment (jeff-dude) [inline models/layerzero/arbitrum/layerzero_arbitrum_send.sql:48]: minor: the format we've been using is:
`<column> >= TIMESTAMP '{{variable}}'`
which casts the string variable to timestamp type

Comment (jeff-dude) [reply models/layerzero/arbitrum/layerzero_arbitrum_send.sql:48]: this also applies on all incremental / non incremental blocks in all spells

File: models/layerzero/base/layerzero_base_send_legacy.sql
Comment (jeff-dude) [inline models/layerzero/base/layerzero_base_send_legacy.sql:1]: general rule:
when building a new spell on dunesql, the legacy version can be simplified to a "dummy" query. here is an example:
https://github.com/duneanalytics/spellbook/blob/main/models/trader_joe/bnb/trader_joe_v2_1_bnb_trades_legacy.sql

you can simply copy/paste that, replace the schema and alias, then call it good for legacy file

note, this applies to all new dunesql spells, both in this PR and future builds until deprecated. we don't want to introduce new code on the legacy engine, but will let existing persist for now.

File: models/layerzero/celo/layerzero_celo_send_legacy.sql
Comment (jeff-dude) [inline models/layerzero/celo/layerzero_celo_send_legacy.sql:1]: same idea here

Comment (jeff-dude) [inline models/layerzero/celo/layerzero_celo_send_legacy.sql:1]: same idea here

File: models/layerzero/layerzero_chain_list.sql [hunk 3]
```diff
@@ -1,36 +1,45 @@
...
-    ('Metis', 151, '0x9740ff91f1985d8d2b71494ae1a2f723bb3ed9e4'),
-    ('Intain', 152, '0x9740ff91f1985d8d2b71494ae1a2f723bb3ed9e4'),
-    ('CoreDAO', 153, '0x9740ff91f1985d8d2b71494ae1a2f723bb3ed9e4'),
-    ('OKX', 155, '0x9740ff91f1985d8d2b71494ae1a2f723bb3ed9e4'),
-    ('Polygon zkEVM', 158, '0x9740ff91f1985d8d2b71494ae1a2f723bb3ed9e4'),
-    ('zkSync Era', 165, '0x9b896c0e23220469c7ae69cb4bbae391eaa4c8da'),
-    ('Moonriver', 167, '0x7004396c99d5690da76a7c59057c5f3a53e01704')
+    ('Ethereum', 101, 0x66a71dcef29a0ffbdbe3c6a460a3b5bc225cd675),
+    ('BNB Chain', 102, 0x3c2269811836af69497e5f486a85d7316753cf62),
+    ('Avalanche', 106, 0x3c2269811836af69497e5f486a85d7316753cf62),
```
Comment (jeff-dude) [inline models/layerzero/layerzero_chain_list.sql:12]: just a general question, the address value is the same on both chains? this isn't what caused the duplicates?

Comment (jeff-dude) [reply models/layerzero/layerzero_chain_list.sql:12]: i notice addresses are repeated often in this file

**Review by jeff-dude — CHANGES_REQUESTED** [review]
left a few notes below, where some changes apply across all blockchain spells. simplify legacy files.

thanks again for putting this together!

**Comment by jeff-dude** [comment]
https://github.com/duneanalytics/spellbook/actions/runs/5890134216/job/15974640043?pr=4104#step:13:23

this test failure should go away once you get rid of legacy logic.

File: models/layerzero/layerzero_chain_list.sql (hunk 3 above)
Comment (BennyFeng) [reply models/layerzero/layerzero_chain_list.sql:12]: No, they are not duplicates. This Spell is a chain list copy from lz official docs and these `Endpoint` contracts are different blockchains. Also, the uniqueness check the chain id only, not the contract address.

File: models/layerzero/arbitrum/layerzero_arbitrum_send.sql (hunk 1 above)
Comment (BennyFeng) [reply models/layerzero/arbitrum/layerzero_arbitrum_send.sql]: Updated.

File: models/layerzero/arbitrum/layerzero_arbitrum_send.sql (hunk 2 above)
Comment (BennyFeng) [reply models/layerzero/arbitrum/layerzero_arbitrum_send.sql]: Updated.

File: models/layerzero/base/layerzero_base_send_legacy.sql
Comment (BennyFeng) [reply models/layerzero/base/layerzero_base_send_legacy.sql:1]: Updated.

**Review by jeff-dude — APPROVED** [review]
looks good, thanks again!

i will push a final commit in a minute to clean up some of the legacy spark code, then go ahead and get this live on dunesql for ya 🙌

**Pull request merged**